quickwit/quickwit-indexing/src/source/mod.rs

```diff
@@ -369,6 +377,35 @@ impl Handler<SuggestTruncate> for SourceActor {
     }
 }
 
+#[derive(Debug, Default)]
+pub struct BatchBuilder {
+    docs: Vec<Bytes>,
+    num_bytes: u64,
+    checkpoint_delta: SourceCheckpointDelta,
+}
+
+impl BatchBuilder {
+    pub fn build(self) -> RawDocBatch {
+        RawDocBatch {
+            docs: self.docs,
+            checkpoint_delta: self.checkpoint_delta,
+            force_commit: false,
+        }
+    }
+    pub fn build_force(self) -> RawDocBatch {
```

Comment:
This method is never used, right? Forcing a commit can only be done via the ingest source.

Reply:
current code is doing that:
```
        if self.should_exit() {
            info!(subscription = %self.subscription, "Reached end of subscription.");
            ctx.ask(doc_processor_mailbox, batch.build_force())
                .await
                .context("Failed to force commit last batch!")?;
            ctx.send_exit_with_success(doc_processor_mailbox).await?;
            return Err(ActorExitStatus::Success);
        }
```

I think if it is not working, then, I can remove this code, we could exit after we got call back of commit

Reply:
I removed the code (build force). In next PR (at_least once impl), I will ensure we exit only when all msg are ack

Reply:
Sounds good. Thanks.